### PR TITLE
build(test/linting) update test/linting dependencies

### PR DESCRIPTION
**Pull request by kibertoad**
## PR Checklist

- [X] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md

## PR Type
```
[X] Build related changes
```

## What is the new behavior?

Several test/linting dependencies were updated, new husky configuration syntax is used

## Does this PR introduce a breaking change?
```
[X] No
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -106,22 +105,21 @@
     "gulp-sourcemaps": "^2.6.4",
     "gulp-typescript": "^3.1.6",
     "gulp-watch": "^4.3.11",
-    "husky": "^0.14.3",
+    "husky": "^1.3.1",
     "imports-loader": "^0.7.0",
-    "istanbul": "^0.4.5",
```
Comment (kibertoad) [inline package.json]: This shouldn't be needed anymore, `nyc` depends on `istanbul-lib-coverage` instead.

**Comment by kamilmysliwiec** [comment]
We are going to set up [renovate](https://github.com/renovatebot/renovate) soon so updates won't be needed anymore

**Comment by kibertoad** [comment]
@kamilmysliwiec Husky configuration update is still a manual step, though.

**Comment by kamilmysliwiec** [comment]
Oh well, you're right. Thanks!

**Pull request merged**